src/pages/store/cart.tsx

```diff
@@ -42,37 +41,23 @@ enum CartState {
   CONFIRMED,
 }
 
-const clientCartToCookie = (items: ClientCartItem[]): string =>
-  JSON.stringify(
-    items.map(item => ({
-      itemUUID: item.uuid,
-      optionUUID: item.option.uuid,
-      quantity: item.quantity,
-    }))
-  );
-
-const calculateOrderTotal = (cart: ClientCartItem[]): number => {
-  return cart.reduce((total, { quantity, option: { price } }) => total + quantity * price, 0);
-};
-
 const StoreCartPage = ({ user: { credits }, savedCart, pickupEvents }: CartPageProps) => {
   const [cart, setCart] = useState(savedCart);
   const [pickupIndex, setPickupIndex] = useState(0);
   const [cartState, setCartState] = useState(CartState.PRECHECKOUT);
-  const [orderTotal, setOrderTotal] = useState(calculateOrderTotal(savedCart));
+  const [orderTotal, setOrderTotal] = useState(CartService.calculateOrderTotal(savedCart));
   const [liveCredits, setLiveCredits] = useState(credits);
 
   // update the cart cookie and order total
   useEffect(() => {
-    setOrderTotal(calculateOrderTotal(cart));
-    setClientCookie(CookieType.CART, clientCartToCookie(cart));
+    setOrderTotal(CartService.calculateOrderTotal(cart));
   }, [cart, credits]);
 
-  const cartInvalid = useMemo(() => cart.some(item => validateClientCartItem(item)), [cart]);
+  const cartInvalid = useMemo(() => CartService.validateClientCart(cart), [cart]);
```

Comment:
It should list the reasons why it's invalid

![image](https://github.com/acmucsd/membership-portal-ui-v2/assets/22133785/093de38a-ca12-4c4e-b0c8-340e7794d696)

I think `CartService.validateClientCart` should return a list of errors (so no errors = valid) rather than a boolean, so the errors aren't lost